tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,269 @@
+        io_thread = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        io_thread.daemon = True  # Daemonize thread
+        io_thread.start()
+        thr_iam.start()
+        LOGGER.info("Step 2: Started independent background threads for IOs on already created "
+                    "user and IAM user creation/deletion.")
+
+        LOGGER.info("Step 3: Perform soft-failure of %s control pods in loop and check cluster "
+                    "has failures", self.repl_num - 1)
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        event.set()
+        for pod in pod_list:
+            self.node_master_list[0].kill_process_in_container(pod_name=pod,
+                                                               container_name=const.CORTX_CSM_POD,
+                                                               process_name=const.CSM_AGENT_PRC,
+                                                               safe_kill=True)
+            LOGGER.info("Step 3.1: Check cluster has failures.")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 3.1: Checked cluster has failures.")
+        LOGGER.info("Step 3: Performed soft-failure of %s control pods in loop and checked cluster "
+                    "has failures", self.repl_num - 1)
+
+        time.sleep(15)
+        event.clear()
+        LOGGER.info("Waiting for background threads to join")
+        thr_iam.join()
+        io_thread.join()
+        LOGGER.info("Step 4: Verifying responses from background processes")
+        LOGGER.info("Step 4.1: Verify background process for IAM user creation")
+        iam_resp = tuple()
+        while len(iam_resp) != 4:
+            iam_resp = iam_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not iam_resp:
+            assert_utils.assert_true(False, "Background process failed to do IAM user CRUD "
+                                            "operations")
+        failed = iam_resp[1]
+        user_del_failed = iam_resp[2]
+        created_users = iam_resp[3]
+        if failed:
+            assert_utils.assert_true(False, "No IAM user creation/deletion expected to fail before "
+                                            "or after soft-failure: {failed}")
+        if created_users:
+            for i_i in created_users:
+                self.s3_clean.update(i_i)
+        LOGGER.info("Updating dict for clean up with remaining users")
+        for i_i in user_del_failed:
+            self.s3_clean.update({i_i: del_users[i_i]})
+        LOGGER.info("Step 4.1: Verified background process for IAM user creation/deletion")
+
+        LOGGER.info("Step 4.2: Verify background process of IOs")
+        responses = {}
```

Comment:
dict()

Reply:
Same. giving me pylint warning
I think we should you this only
Consider using {} instead of dict()  (use-dict-literal)

Reply:
Better to have dict() as we had discussed this many times before.